### ISSUE TITLE
Question: Can tag order by forced in tree view?

### ISSUE THREAD

**Issue by jeep** [Gruntfuggly/todo-tree#675]
When I have: `"todo-tree.tree.groupedByTag": true,` The order of the groups in the treeview seem random. I would like to have them in a specific order (not alphabetical). The settings help text seems to indicate that they should appear in the order they are defined in: `todo-tree.general.tags`

Even when I remove tag groups, the order seems arbitrary. Ideally, I'd like everything to sort in the order that the tagGroups are defined. If that isn't possible, then I'd like them to show up in the order that the tags are defined. I am not seeing any settings to help me control that. Is it something I can control?

Thanks.

**Comment by Elias-Graf** [comment]
It seems that the order of `todo-tree.general.tags` is working for the "flat view". The tree is ordered the same as the files in the file system. The items itself seem to be in order, that are defined in.

I've never had any issues with this / noticed this. But I've also never had more than around 10 markers in a single document.

**Comment by jeep** [comment]
Right, tree view appears to be in random order. I didn't touch a todo, but edited a file with a TODO in it and the order of keywords changed.

**Comment by maxaxehax** [comment]
Glad this is marked as a [bug] which can hopefully be tackled in the near future.
While the extension is already awesome, the idea to be able to have a consistent sorting of tag-groups would be really excellent if you want to use the groups for priorization of open work to make a junior team internalize prioritization of work as engineering best-practices, as I am trying to do:

0 bugfixing if there are open defects
1 implementing pending tests
2 implementing current feature
3 discussing, clarifying and reviewing code (to maximize knowledge transfer & upskilling)
4 removing hacks, workarounds and rushed implementations
5 refactoring code smells
6 learning new things by doing experimental changes

Without a consistent sorting of TODO groups, they get mixed up all the time on which todos they should prioritize over others, and the whole learning aspect / developing an instinct for eXtreme Programming mindeset is sadly lost.

I can confirm the sorting is not affected by any kind of lexicographic nor file-order factor - I've tested it extensively. It would be beautiful for my (admittedly niche and highly-specific, but valuable) use case.

**Comment by FanaticPythoner** [comment]
@jeep @Elias-Graf @maxaxehax See: https://github.com/FanaticPythoner/better-todo-tree/commit/efcf97243292d5b6839e2616cd76404b521048a3
